doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/user-guides/autoscaling.md

```diff
@@ -0,0 +1,91 @@
+while the Ray application is running. On the other hand, if you pre-provision a fixed
+number of Ray workers, all of the Ray workers can be started in parallel, potentially reducing your application's
+runtime.
+
+**Resource usage.** The autoscaler itself consumes CPU and memory resources.
+These resources must be accounted for when considering the Ray head pod's scheduling.
+The default autoscaler container resources are .5 CPU requests and limits, 512 megabytes memory request and limits.
+See the {ref}`configuration guide<kuberay-autoscaling-config>` for details.
+
+## Ray Autoscaler vs. other autoscalers
+We describe the relationship between the Ray autoscaler and other autoscalers in the Kubernetes
+ecosystem.
+
+### Ray Autoscaler vs. Horizontal Pod Autoscaler
+The Ray autoscaler adjusts the number of Ray nodes in a Ray cluster.
+On Kubernetes, each Ray node is run as a Kubernetes pod. Thus in the context of Kubernetes,
+the Ray autoscaler scales Ray **pod quantities**. In this sense, the Ray autoscaler
+plays a role similar to that of the Kubernetes
+[Horizontal Pod Autoscaler](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) (HPA).
+However, the following features distinguish the Ray Autoscaler from the HPA.
+## Load metrics are based on application semantics
+The Horizontal Pod Autoscaler determines scale based on physical usage metrics like CPU
+and memory. By contrast, the Ray autoscaler uses the logical resources expressed in
+task and actor annotations. For instance, if each Ray container spec in your RayCluster CR indicates
+a limit of 10 CPUs, and you submit twenty tasks annotated with `@ray.remote(num_cpus=5)`,
+10 Ray pods will be created to satisfy the 100-CPU resource demand.
+In this respect, the Ray autoscaler is similar to the
+[Kuberentes Cluster Autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler),
+which makes scaling decisions based on the logical resources expressed in container
+resource requests.
+## Fine-grained control of scale-down
+To accommodate the statefulness of Ray applications, the Ray autoscaler has more
+fine-grained control over scale-down than the Horizontal Pod Autoscaler. In addition to
+determining desired scale, the Ray Autoscaler is able to select precisely which pods
+to scale down. The KubeRay operator then deletes that pod.
+By contrast, the Horizontal Pod Autoscaler can only decrease a replica count, without much
+control over which pods are deleted. For a Ray application, downscaling a random
+pod could be dangerous.
+## Architecture: One Ray Autoscaler per Ray Cluster.
+Horizontal Pod Autoscaling is centrally controlled by a manager in the Kubernetes control plane;
+the manager controls the scale of many Kubernetes objects.
+By contrast, each Ray cluster is managed by its own Ray autoscaler process,
+running as a sidecar container in the Ray head pod. This design choice is motivated
+by considerations of scalability.
```

Comment:
and co-versioning the autoscaler with Ray, could also mention that

Reply:
mentioned